resolver/src/main/java/io/netty/resolver/HostsFileParser.java

```diff
@@ -153,14 +157,20 @@ private static File locateHostsFile() {
                 for (int i = 1; i < lineParts.size(); i ++) {
                     String hostname = lineParts.get(i);
                     String hostnameLower = hostname.toLowerCase(Locale.ENGLISH);
-                    if (!entries.containsKey(hostnameLower)) {
-                        // trying to map a host to multiple IPs is wrong
-                        // only the first entry is honored
-                        entries.put(hostnameLower, InetAddress.getByAddress(hostname, ipBytes));
+                    InetAddress address = InetAddress.getByAddress(hostname, ipBytes);
+                    if (address instanceof Inet4Address) {
+                        // Map::putIfAbsent is Java8 :(
```

Comment:
it is sad but we can remove this comment :)

We could also optimistically put and restore ... although I don't think it is a big deal here either way.